Travis failure seems very similar to https://github.com/cilium/cilium/issues/8596. Is this related to the recent work in kvstore package? /cc @raybejjani @aanm 

```
=== RUN   Test

{"level":"warn","ts":"2020-02-27T20:56:56.379Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-0e6684c1-b7f8-461f-8154-04a91e2b4c8a/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:56:56.379Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-0e6684c1-b7f8-461f-8154-04a91e2b4c8a/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:56:56.393Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-90c608fe-34b3-4784-a7bf-c15f405dd802/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:56:56.393Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-90c608fe-34b3-4784-a7bf-c15f405dd802/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:56:56.413Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-56f8f86a-5c9f-4050-a3ac-bccffca8856c/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:56:56.413Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-56f8f86a-5c9f-4050-a3ac-bccffca8856c/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:57:56.442Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-76ed2f92-6d3b-4712-8c40-8ccd86996130/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:57:56.442Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-76ed2f92-6d3b-4712-8c40-8ccd86996130/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:57:56.463Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-9e086f91-d94c-41f8-8dc7-ea2d58521e19/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:57:56.466Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-9e086f91-d94c-41f8-8dc7-ea2d58521e19/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

{"level":"warn","ts":"2020-02-27T20:57:56.466Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-9e086f91-d94c-41f8-8dc7-ea2d58521e19/127.0.0.1:4002","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}

----------------------------------------------------------------------

FAIL: <autogenerated>:1: StoreConsulSuite.TestStoreLocalKeyProtection

store_test.go:270:

    c.Assert(expect(func() bool {

        v, err := kvstore.Client().Get(context.TODO(), store.keyPath(&localKey1))

        return err == nil && v == nil

    }), IsNil)

... value *errors.errorString = &errors.errorString{s:"timeout while waiting for expected value"} ("timeout while waiting for expected value")

OOPS: 9 passed, 1 FAILED

--- FAIL: Test (337.74s)

FAIL

coverage: 9.5% of statements in ./...

FAIL	github.com/cilium/cilium/pkg/kvstore/store	337.780s
```

--- a/Documentation/cmdref/cilium-agent.md
+++ b/Documentation/cmdref/cilium-agent.md
@@ -133,6 +133,7 @@ cilium-agent [flags]
       --nat46-range string                            IPv6 prefix to map IPv4 addresses to (default "0:0:0:0:0:FFFF::/96")
       --node-port-mode string                         BPF NodePort mode ("snat", "dsr", "hybrid") (default "hybrid")
       --node-port-range strings                       Set the min/max NodePort port range (default [30000,32767])
+      --policy-audit-mode                             Enable policy audit (non-drop) mode
       --policy-queue-size int                         size of queues for policy-related events (default 100)
       --pprof                                         Enable serving the pprof debugging API
       --preallocate-bpf-maps                          Enable BPF map pre-allocation (default true)

--- a/bpf/lib/policy.h
+++ b/bpf/lib/policy.h
@@ -189,8 +189,10 @@ policy_can_access_ingress(struct __ctx_buff *ctx, __u32 src_identity,
 
 	cilium_dbg(ctx, DBG_POLICY_DENIED, src_identity, SECLABEL);
 
-#ifdef IGNORE_DROP
-	ret = CTX_ACT_OK;
+#ifdef POLICY_AUDIT_MODE
+	if (IS_ERR(ret)) {
+		ret = CTX_ACT_OK;
+	}
 #endif
 
 	return ret;
@@ -223,8 +225,10 @@ policy_can_egress(struct __ctx_buff *ctx, __u32 identity, __u16 dport, __u8 prot
 
 	cilium_dbg(ctx, DBG_POLICY_DENIED, SECLABEL, identity);
 
-#ifdef IGNORE_DROP
-	ret = CTX_ACT_OK;
+#ifdef POLICY_AUDIT_MODE
+	if (IS_ERR(ret)) {
+		ret = CTX_ACT_OK;
+	}
 #endif
 
 	return ret;

--- a/daemon/daemon_main.go
+++ b/daemon/daemon_main.go
@@ -728,6 +728,9 @@ func init() {
 	flags.Bool(option.DisableCNPStatusUpdates, false, `Do not send CNP NodeStatus updates to the Kubernetes api-server (recommended to run with "cnp-node-status-gc=false" in cilium-operator)`)
 	option.BindEnv(option.DisableCNPStatusUpdates)
 
+	flags.Bool(option.PolicyAuditModeArg, false, "Enable policy audit (non-drop) mode")
+	option.BindEnv(option.PolicyAuditModeArg)
+
 	viper.BindPFlags(flags)
 }
 
@@ -900,6 +903,7 @@ func initEnv(cmd *cobra.Command) {
 	option.Config.Opts.SetBool(option.Conntrack, !option.Config.DisableConntrack)
 	option.Config.Opts.SetBool(option.ConntrackAccounting, !option.Config.DisableConntrack)
 	option.Config.Opts.SetBool(option.ConntrackLocal, false)
+	option.Config.Opts.SetBool(option.PolicyAuditMode, option.Config.PolicyAuditMode)
 
 	monitorAggregationLevel, err := option.ParseMonitorAggregationLevel(option.Config.MonitorAggregation)
 	if err != nil {

--- a/install/kubernetes/cilium/charts/config/templates/configmap.yaml
+++ b/install/kubernetes/cilium/charts/config/templates/configmap.yaml
@@ -335,6 +335,7 @@ data:
   enable-remote-node-identity: {{ .Values.global.remoteNodeIdentity | quote }}
 
   synchronize-k8s-nodes: {{ .Values.global.synchronizeK8sNodes | quote }}
+  policy-audit-mode: {{ .Values.global.policyAuditMode | quote }}
 
 {{- if .Values.global.ipv4.enabled }}
   operator-api-serve-addr: '127.0.0.1:9234'

--- a/install/kubernetes/cilium/values.yaml
+++ b/install/kubernetes/cilium/values.yaml
@@ -354,3 +354,8 @@ global:
   # psp creates and binds PodSecurityPolicies for the components that require it
   psp:
     enabled: false
+
+  # enables non-drop mode for installed policies. In audit mode
+  # packets affected by policies will not be dropped. Policy related
+  # decisions can be checked via the poicy verdict messages.
+  policyAuditMode: false

--- a/pkg/datapath/linux/config/config.go
+++ b/pkg/datapath/linux/config/config.go
@@ -274,6 +274,10 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		ctmap.WriteBPFMacros(fw, nil)
 	}
 
+	if option.Config.PolicyAuditMode {
+		cDefinesMap["POLICY_AUDIT_MODE"] = "1"
+	}
+
 	// Since golang maps are unordered, we sort the keys in the map
 	// to get a consistent writtern format to the writer. This maintains
 	// the consistency when we try to calculate hash for a datapath after

--- a/pkg/option/config.go
+++ b/pkg/option/config.go
@@ -756,6 +756,9 @@ const (
 
 	// AzureResourceGroup is the resource group of the nodes used for the cluster
 	AzureResourceGroup = "azure-resource-group"
+
+	// PolicyAuditModeArg argument enables policy audit mode.
+	PolicyAuditModeArg = "policy-audit-mode"
 )
 
 // Default string arguments
@@ -1527,6 +1530,11 @@ type DaemonConfig struct {
 
 	// AzureResourceGroup is the resource group of the nodes used for the cluster
 	AzureResourceGroup string
+
+	// PolicyAuditMode enables non-drop mode for installed policies. In
+	// audit mode packets affected by policies will not be dropped.
+	// Policy related decisions can be checked via the poicy verdict messages.
+	PolicyAuditMode bool
 }
 
 var (
@@ -1984,6 +1992,7 @@ func (c *DaemonConfig) Populate() {
 	c.CTMapEntriesTimeoutSVCAny = viper.GetDuration(CTMapEntriesTimeoutSVCAnyName)
 	c.CTMapEntriesTimeoutSYN = viper.GetDuration(CTMapEntriesTimeoutSYNName)
 	c.CTMapEntriesTimeoutFIN = viper.GetDuration(CTMapEntriesTimeoutFINName)
+	c.PolicyAuditMode = viper.GetBool(PolicyAuditModeArg)
 
 	if nativeCIDR := viper.GetString(IPv4NativeRoutingCIDR); nativeCIDR != "" {
 		c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)

--- a/pkg/option/daemon.go
+++ b/pkg/option/daemon.go
@@ -34,6 +34,7 @@ var (
 		DropNotify:          &specDropNotify,
 		TraceNotify:         &specTraceNotify,
 		PolicyVerdictNotify: &specPolicyVerdictNotify,
+		PolicyAuditMode:     &specPolicyAuditMode,
 		MonitorAggregation:  &specMonitorAggregation,
 		NAT46:               &specNAT46,
 	}

--- a/pkg/option/endpoint.go
+++ b/pkg/option/endpoint.go
@@ -24,6 +24,7 @@ var (
 		DropNotify:          &specDropNotify,
 		TraceNotify:         &specTraceNotify,
 		PolicyVerdictNotify: &specPolicyVerdictNotify,
+		PolicyAuditMode:     &specPolicyAuditMode,
 		MonitorAggregation:  &specMonitorAggregation,
 		NAT46:               &specNAT46,
 	}

--- a/pkg/option/runtime_options.go
+++ b/pkg/option/runtime_options.go
@@ -28,6 +28,7 @@ const (
 	DropNotify          = "DropNotification"
 	TraceNotify         = "TraceNotification"
 	PolicyVerdictNotify = "PolicyVerdictNotification"
+	PolicyAuditMode     = "PolicyAuditMode"
 	MonitorAggregation  = "MonitorAggregationLevel"
 	NAT46               = "NAT46"
 	AlwaysEnforce       = "always"
@@ -82,6 +83,11 @@ var (
 	specPolicyVerdictNotify = Option{
 		Define:      "POLICY_VERDICT_NOTIFY",
 		Description: "Enable policy verdict notifications",
+	}
+
+	specPolicyAuditMode = Option{
+		Define:      "POLICY_AUDIT_MODE",
+		Description: "Enable audit mode for policies",
 	}
 
 	specMonitorAggregation = Option{

--- a/test/runtime/Policies.go
+++ b/test/runtime/Policies.go
@@ -1410,64 +1410,108 @@ var _ = Describe("RuntimePolicies", func() {
 			vm.ContainerRm(initContainer).ExpectSuccess("Container initContainer cannot be deleted")
 		})
 
-		It("Init Ingress Policy Default Drop Test", func() {
-			By("Starting cilium monitor in background")
-			ctx, cancel := context.WithCancel(context.Background())
-			monitorRes := vm.ExecInBackground(ctx, "cilium monitor --type drop --type trace")
-			defer cancel()
-
-			By("Creating an endpoint")
-			res := vm.ContainerCreate(initContainer, constants.NetperfImage, helpers.CiliumDockerNetwork, "-l somelabel")
+		createEndpoint := func(cmdArgs ...string) (endpointID string, endpointIP *models.AddressPair) {
+			res := vm.ContainerCreate(initContainer, constants.NetperfImage, helpers.CiliumDockerNetwork, "-l somelabel", cmdArgs...)
 			res.ExpectSuccess("Failed to create container")
 
 			endpoints, err := vm.GetAllEndpointsIds()
 			Expect(err).Should(BeNil(), "Unable to get IDs of endpoints")
-			endpointID, exists := endpoints[initContainer]
+			var exists bool
+			endpointID, exists = endpoints[initContainer]
 			Expect(exists).To(BeTrue(), "Expected endpoint ID to exist for %s", initContainer)
 			ingressEpModel := vm.EndpointGet(endpointID)
 			Expect(ingressEpModel).NotTo(BeNil(), "nil model returned for endpoint %s", endpointID)
 
-			endpointIP := ingressEpModel.Status.Networking.Addressing[0]
+			endpointIP = ingressEpModel.Status.Networking.Addressing[0]
+			return
+		}
+
+		It("tests ingress", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			By("Starting cilium monitor in background")
+			monitorRes := vm.ExecInBackground(ctx, "cilium monitor --type drop --type trace")
+			defer cancel()
+
+			By("Creating an endpoint")
+			endpointID, endpointIP := createEndpoint()
 
 			// Normally, we start pinging fast enough that the endpoint still has identity "init" / 5,
 			// and we continue pinging as the endpoint changes its identity for label "somelabel".
 			// So these pings will be dropped by the policies for both identity 5 and the new identity
 			// for label "somelabel".
 			By("Testing ingress with ping from host to endpoint")
-			res = vm.Exec(helpers.Ping(endpointIP.IPV4))
+			res := vm.Exec(helpers.Ping(endpointIP.IPV4))
 			res.ExpectFail("Unexpectedly able to ping endpoint with no ingress policy")
 
 			By("Testing cilium monitor output")
-			err = monitorRes.WaitUntilMatch("xx drop (Policy denied")
+			err := monitorRes.WaitUntilMatch("xx drop (Policy denied")
 			Expect(err).To(BeNil(), "Default drop on ingress failed")
 			monitorRes.ExpectDoesNotContain(fmt.Sprintf("-> endpoint %s ", endpointID),
 				"Unexpected ingress traffic to endpoint")
 		})
 
-		It("Init Egress Policy Default Drop Test", func() {
-			hostIP := "10.0.2.15"
-
+		It("tests egress", func() {
 			By("Starting cilium monitor in background")
 			ctx, cancel := context.WithCancel(context.Background())
 			monitorRes := vm.ExecInBackground(ctx, "cilium monitor --type drop --type trace")
 			defer cancel()
 
 			By("Creating an endpoint")
-			res := vm.ContainerCreate(initContainer, constants.NetperfImage, helpers.CiliumDockerNetwork, "-l somelabel", "ping", hostIP)
-			res.ExpectSuccess("Failed to create container")
-
-			endpoints, err := vm.GetAllEndpointsIds()
-			Expect(err).To(BeNil(), "Unable to get IDs of endpoints")
-			endpointID, exists := endpoints[initContainer]
-			Expect(exists).To(BeTrue(), "Expected endpoint ID to exist for %s", initContainer)
-			egressEpModel := vm.EndpointGet(endpointID)
-			Expect(egressEpModel).NotTo(BeNil(), "nil model returned for endpoint %s", endpointID)
+			endpointID, _ := createEndpoint("ping", "10.0.2.15")
 
 			By("Testing cilium monitor output")
-			err = monitorRes.WaitUntilMatch("xx drop (Policy denied")
+			err := monitorRes.WaitUntilMatch("xx drop (Policy denied")
 			Expect(err).To(BeNil(), "Default drop on egress failed")
 			monitorRes.ExpectDoesNotContain(fmt.Sprintf("-> endpoint %s ", endpointID),
 				"Unexpected reply traffic to endpoint")
+		})
+
+		Context("With PolicyAuditMode", func() {
+			BeforeEach(func() {
+				vm.ExecCilium("config PolicyAuditMode=Enabled").ExpectSuccess("unable to change daemon configuration")
+			})
+
+			It("tests ingress", func() {
+				ctx, cancel := context.WithCancel(context.Background())
+				By("Starting cilium monitor in background")
+				monitorRes := vm.ExecInBackground(ctx, "cilium monitor --type drop --type trace --type policy-verdict")
+				defer cancel()
+
+				By("Creating an endpoint")
+				endpointID, endpointIP := createEndpoint()
+
+				By("Testing ingress with ping from host to endpoint")
+				res := vm.Exec(helpers.Ping(endpointIP.IPV4))
+				res.ExpectSuccess("Not able to ping endpoint with no ingress policy")
+
+				By("Testing cilium monitor output")
+				err := monitorRes.WaitUntilMatch("Policy verdict log")
+				Expect(err).To(BeNil(), "Default policy verdict on ingress failed")
+				monitorRes.ExpectContains(
+					fmt.Sprintf("local EP ID %s, remote ID 1, dst port 0, proto 1, ingress true, action allow", endpointID),
+					"No ingress policy log record",
+				)
+				monitorRes.ExpectContains(fmt.Sprintf("-> endpoint %s ", endpointID), "No ingress traffic to endpoint")
+			})
+
+			It("tests egress", func() {
+				By("Starting cilium monitor in background")
+				ctx, cancel := context.WithCancel(context.Background())
+				monitorRes := vm.ExecInBackground(ctx, "cilium monitor --type drop --type trace --type policy-verdict")
+				defer cancel()
+
+				By("Creating an endpoint")
+				endpointID, _ := createEndpoint("ping", "10.0.2.15")
+
+				By("Testing cilium monitor output")
+				err := monitorRes.WaitUntilMatch("Policy verdict log")
+				Expect(err).To(BeNil(), "Default policy verdict on egress failed")
+				monitorRes.ExpectContains(
+					fmt.Sprintf("ID %s, remote ID 1, dst port 0, proto 1, ingress false, action allow", endpointID),
+					"No egress policy log record",
+				)
+				monitorRes.ExpectContains(fmt.Sprintf("-> endpoint %s ", endpointID), "No reply traffic to endpoint")
+			})
 		})
 	})
 	Context("Init Policy Test", func() {

